lib/streamlit/web/server/server.py

```diff
@@ -299,14 +305,24 @@ def _create_app(self) -> tornado.web.Application:
             (
                 make_url_path_regex(
                     base,
-                    UPLOAD_FILE_ROUTE,
+                    rf"{UPLOAD_FILE_ENDPOINT}/(?P<session_id>[^/]+)/(?P<file_id>[^/]+)",
                 ),
                 UploadFileRequestHandler,
                 dict(
                     file_mgr=self._runtime.uploaded_file_mgr,
                     is_active_session=self._runtime.is_active_session,
                 ),
             ),
+            (
+                make_url_path_regex(
+                    base,
+                    UPLOAD_FILE_URLS_ROUTE,
+                ),
+                UploadFileUrlsRequestHandler,
+                dict(
+                    file_mgr=self._runtime.uploaded_file_mgr,
+                ),
+            ),
```

Comment:
same comment about having a TODO about potentially removing this later so that we don't miss it